app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/DeleteRecordController.scala

```diff
@@ -16,32 +16,58 @@
+import play.api.mvc.{Action, AnyContent, ControllerComponents, Request}
+import uk.gov.hmrc.http.{HeaderCarrier, UpstreamErrorResponse}
 import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
 import uk.gov.hmrc.tradergoodsprofilesdatastore.connectors.RouterConnector
-import uk.gov.hmrc.tradergoodsprofilesdatastore.controllers.actions.IdentifierAction
+import uk.gov.hmrc.tradergoodsprofilesdatastore.controllers.actions.{IdentifierAction, StoreLatestAction}
+import uk.gov.hmrc.tradergoodsprofilesdatastore.repositories.RecordsRepository
 
 import javax.inject.Inject
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success}
 
 class DeleteRecordController @Inject() (
   cc: ControllerComponents,
   routerConnector: RouterConnector,
-  identify: IdentifierAction
+  recordsRepository: RecordsRepository,
+  identify: IdentifierAction,
+  storeLatestAction: StoreLatestAction
 )(implicit ec: ExecutionContext)
     extends BackendController(cc)
     with Logging {
 
-  def deleteRecord(eori: String, recordId: String): Action[AnyContent] = identify.async { implicit request =>
-    routerConnector.deleteRecord(eori, recordId).map(response => response) transform {
-      case Success(_)                                                             => Success(NoContent)
-      case Failure(cause: UpstreamErrorResponse) if cause.statusCode == NOT_FOUND => Success(NotFound)
-      case Failure(cause: UpstreamErrorResponse)                                  =>
-        logger.error(s"Deleted record failed with ${cause.statusCode} with message: ${cause.message}")
-        Success(InternalServerError)
-    }
+  def deleteRecord(eori: String, recordId: String): Action[AnyContent] = (identify andThen storeLatestAction).async {
+    implicit request =>
+      isRecordActive(eori, recordId).flatMap {
+        case true  =>
+          makeRecordInactive(eori, recordId) transform {
+            case Success(_)                            => Success(NoContent)
+            case Failure(cause: UpstreamErrorResponse) =>
+              logger.error(s"Deleted record failed with ${cause.statusCode} with message: ${cause.message}")
+              Success(InternalServerError)
+          }
+        case false =>
+          makeRecordInactive(eori, recordId) transform {
+            case Success(_)                                                               => Success(NotFound)
+            case Failure(cause: UpstreamErrorResponse) if cause.statusCode == BAD_REQUEST => Success(NotFound)
+            case Failure(cause: UpstreamErrorResponse)                                    =>
+              logger.error(s"Deleted record failed with ${cause.statusCode} with message: ${cause.message}")
+              Success(InternalServerError)
+          }
+      }
   }
+
+  private def makeRecordInactive(eori: String, recordId: String)(implicit
+    hc: HeaderCarrier
+  ): Future[Done] =
```

Comment:
this doesn't do anything` .map(response => response)`

Reply:
what's the correct way of righting it then? It needs to get passed to the transform

Reply:
just leave out the map altogether as the deleteRecord is already returning a Future[Done] :)